WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Bulk Edit Price/BulkUpdatePriceSettingsViewModel.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import Yosemite
+
+/// View Model logic for the bulk price setting screen
+///
+final class BulkUpdatePriceSettingsViewModel {
+    /// Represents the possible states for the save button.
+    enum ButtonState: Equatable {
+        case enabled
+        case disabled
+        case loading
+    }
+
+    /// Represents the possible errors during the bulk update
+    enum BulkUpdatePriceError: Error, Equatable {
+        case inputValidationError(ProductPriceSettingsError)
+        case priceUpdateError
+    }
+
+    /// Indicates what price we are editing
+    ///
+    enum EditingPriceType {
+        case regular
+        case sale
+
+        func keyPathForPriceType() -> KeyPath<ProductVariation, String?> {
+            switch self {
+            case .regular:
+                return \.regularPrice
+            case .sale:
+                return \.salePrice
+            }
+        }
+    }
+
+    /// The state of save price setting button
+    @Published private(set) var saveButtonState: ButtonState = .disabled
+
+    /// The error state
+    @Published private(set) var bulkUpdatePriceError: BulkUpdatePriceError? = nil
+
+    /// A Closure to be called when the price update is successful
+    private let priceUpdateDidFinish: () -> Void
+
+    private var currentPrice: String? = nil
```

Comment:
Could we add some docs on why we are storing this price & where it is used? eg: for validating prices in real-time.

Reply:
Basically this holds the latest entered price, it is used on the save button action to validate the price and the create the new array of variations with updated price for the bulk update action.
I updated the documentation!